rskj-core/src/main/java/org/ethereum/rpc/FilterManager.java

```diff
@@ -71,6 +72,11 @@ public int registerFilter(Filter filter) {
 
     public boolean removeFilter(int id) {
         synchronized (filterLock) {
+
+            Filter filter = installedFilters.get(id);
+            if (filter == null) {

```

Comment:
```suggestion
            if (installedFilters.remove(id) == null) {

```

`installedFilters.remove(id)` already returns a value removed, so I guess you could check it instead of looking for it in `installedFilters`